keras/backend/theano_backend.py

```diff
@@ -386,7 +386,7 @@ def normalize_batch_in_training(x, gamma, beta,
 def batch_normalization(x, mean, var, beta, gamma, epsilon=0.0001):
     '''Apply batch normalization on x given mean, var, beta and gamma.
     '''
-    if theano.config.device.startswith('cuda') or theano.config.device.startswith('gpu'):
+    if x.ndim < 5 and (theano.config.device.startswith('cuda') or theano.config.device.startswith('gpu')):
```

Comment:
In that case shouldn't it be `if x.ndim == 4`?

Also this line is becoming too long, I'd suggest creating a `use_cudnn` intermediate variable.